### PR TITLE
Fix broken windows build

### PR DESCRIPTION
**Pull request by sgpearse**
Fixes our windows build!

### CODE REVIEW

File: apps/vaporgui/MainForm.cpp [hunk 1]
```diff
@@ -1186,10 +1186,10 @@ void MainForm::createBookmark()
     int  customViewportWidth = vpp->GetValueLong(vpp->CustomFramebufferWidthTag, 0);
     int  customViewportHeight = vpp->GetValueLong(vpp->CustomFramebufferHeightTag, 0);
 
-    const int     iconSize = BookmarkParams::DefaultIconSize();
-    const int     iconDataSize = iconSize * iconSize * 3;
-    unsigned char iconData[iconDataSize];
-    char          iconDataString[64];
+    const int      iconSize = BookmarkParams::DefaultIconSize();
+    const int      iconDataSize = iconSize * iconSize * 3;
+    unsigned char *iconData = new unsigned char[iconDataSize];
```
Comment (shaomeng) [inline apps/vaporgui/MainForm.cpp:1191]: If this pointer is already pointing to a memory block, then this line will cause the old block to leak.

**Review by shaomeng — CHANGES_REQUESTED** [review]

File: apps/vaporgui/MainForm.cpp (hunk 1 above)
Comment (shaomeng) [reply apps/vaporgui/MainForm.cpp:1191]: suggest avoid managing memory manually by using a unique ptr:
`auto iconData = std::make_unique<unsigned char[]>( iconDataSize );`

Comment (StasJ) [reply apps/vaporgui/MainForm.cpp]: How could this pointer already be pointing to a memory block?

File: apps/vaporgui/MainForm.cpp [hunk 2]
```diff
@@ -1209,6 +1210,8 @@ void MainForm::createBookmark()
     Base16StringStream is;
     is.write((char *)iconData, iconDataSize);
 
+    if (iconData != nullptr) delete[] iconData;
```
Comment (StasJ) [inline apps/vaporgui/MainForm.cpp:1213]: You can't delete a unique ptr

**Review by StasJ — CHANGES_REQUESTED** [review]
I know that I already made some of these fixes before so I wonder what happened to them... We've had windows building since the OSPRay PR and the Bookmarks PR and but that wouldn't have been possible without the changes here so they must've gotten rolled back somehow.

File: apps/vaporgui/MainForm.cpp (hunk 1 above)
Comment (shaomeng) [reply apps/vaporgui/MainForm.cpp]: > How could this pointer already be pointing to a memory block?

That's right, it cannot. The original code won't produce a memory leak.

**Comment by sgpearse** [comment]
@stasj, regarding your comments:

>  know that I already made some of these fixes before so I wonder what happened to them...

1) I recall fixing some of these problems too.  One of the lines breaking the windows build is in PGroup.cpp:6.  On main, this was last modified by StasJ in October 2020.  This change was before our 3.3 release , but now it's breaking the windows build.

2) Could you please fix the errors regarding your code, with whatever data structures suitable?  I need to focus on #2704, and this code is under your ownership.

**Review by shaomeng — APPROVED** [review]

**Review by StasJ — APPROVED** [review]

**Pull request merged**